src/db/sysdb_ipnetworks.c

```diff
@@ -0,0 +1,439 @@
+/*
+    SSSD
+
+    Authors:
+        Samuel Cabrero <scabrero@suse.com>
+
+    Copyright (C) 2020 SUSE LINUX GmbH, Nuernberg, Germany.
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <arpa/inet.h>
+
+#include "db/sysdb.h"
+#include "db/sysdb_private.h"
+#include "db/sysdb_ipnetworks.h"
+
+errno_t sysdb_getipnetworkbyname(TALLOC_CTX *mem_ctx,
+                                 struct sss_domain_info *domain,
+                                 const char *name,
+                                 struct ldb_result **_res)
+{
+    errno_t ret;
+    TALLOC_CTX *tmp_ctx;
+    static const char *attrs[] = {
+        SYSDB_NAME,
+        SYSDB_NAME_ALIAS,
+        SYSDB_IP_NETWORK_ATTR_NUMBER,
+        SYSDB_DEFAULT_ATTRS,
+        NULL,
+    };
+    char *sanitized_name;
+    char *subfilter;
+    struct ldb_result *res = NULL;
+    struct ldb_message **msgs;
+    size_t msgs_count;
+
+    DEBUG(SSSDBG_TRACE_FUNC, "Searching network by name [%s] in domain [%s]\n",
+          name, domain->name);
+
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        return ENOMEM;
+    }
+
+    ret = sss_filter_sanitize(tmp_ctx, name, &sanitized_name);
+    if (ret != EOK) {
+        goto done;
+    }
+
+    subfilter = talloc_asprintf(tmp_ctx, SYSDB_IP_NETWORK_BYNAME_SUBFILTER,
+                                sanitized_name, sanitized_name);
+    if (subfilter == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
+
+    ret = sysdb_search_ipnetworks(mem_ctx, domain, subfilter,
+                                  attrs, &msgs_count, &msgs);
+    if (ret == EOK) {
+        res = talloc_zero(mem_ctx, struct ldb_result);
+        if (res == NULL) {
+            ret = ENOMEM;
+            goto done;
+        }
+        res->count = msgs_count;
+        res->msgs = talloc_steal(res, msgs);
+    }
+
+    *_res = res;
```

Comment:
Better use `tmp_ctx` everywhere and then here call `*_res = talloc_steal(mem_ctx, res)`. Same for other functions.